File: social-service/src/main/java/com/eirsteir/coffeewithme/social/domain/university/Campus.java
package com.eirsteir.coffeewithme.social.domain.university;

import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.*;
import lombok.Data;
import lombok.ToString;

@Data
@Entity
public class Campus {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  private String name;

  @ManyToOne
  @JoinColumn(name = "university_id")
  @ToString.Exclude
  @JsonIgnore
  private University university;
}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/domain/university/Campus.kt
package com.eirsteir.coffeewithme.social.domain.university

import com.fasterxml.jackson.annotation.JsonIgnore
import lombok.Data
import lombok.ToString
import javax.persistence.*

@Entity
data class Campus(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long? = null,
    val name: String? = null,
    @ManyToOne
    @JoinColumn(name = "university_id")
    @ToString.Exclude
    @JsonIgnore
    var university: University? = null,
)